llvm/tools/gold/gold-plugin.cpp

```diff
@@ -1113,6 +1131,19 @@ static ld_plugin_status allSymbolsReadHook() {
   if (unsigned NumOpts = options::extra.size())
     cl::ParseCommandLineOptions(NumOpts, &options::extra[0]);
 
+  // Initialize time trace profiler
+  if (!options::time_trace_file.empty())
+    llvm::timeTraceProfilerInitialize(options::time_trace_granularity,
+                                      options::extra.size() ? options::extra[0]
+                                                            : "LLVMgold");
```

Comment:
At https://github.com/llvm/llvm-project/blob/6cd86d0fae8cbb752a713860f131b9b759b2cbb8/llvm/tools/gold/gold-plugin.cpp#L316 we set extra[0] to "LLVMgold" if extra is empty, so this shouldn't be needed (extra should never be empty and extra[0] can always be used).

Reply:
I think we reach this line (line 316) only if there are additional `-plugin-opt` that LLVMgold doesn't recognize. If there aren't any of those, I think `options::extra` will remain empty.

Reply:
got it thanks